pkg/controllers/startup_policy.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+)
+
+// replicatedJobsStarted tells if the replicated jobSet is ready
+// or it has finished
+// If replicas is equal then we assume that replicatedJob is considered ready
+func replicatedJobsStarted(replicas int32, rjJobStatus jobset.ReplicatedJobStatus) bool {
+	return replicas == rjJobStatus.Failed+rjJobStatus.Ready+rjJobStatus.Succeeded
+}
+
+func inOrderStartupPolicy(sp *jobset.StartupPolicy) bool {
+	return sp != nil && sp.StartupPolicyOrder == jobset.InOrder
+}
+
+// generateStartupPolicyCondition generates the StartupPolicyCondition
+// based on policy
+// We use startupPolicyCompleted or startupPolicyInProgress
+func generateStartupPolicyCondition(policy startupPolicyCondition) metav1.Condition {
```

Comment:
the startupPolicyCondition is a redundant type, I suggest we do the following:

```func generateCompletedStartupPolicyCondition(status metav1.ConditionStatus)```

Reply:
good call. Updated.